pkg/api/v1beta1/types.go

```diff
@@ -397,6 +397,23 @@ type EnvVar struct {
 	Key  string `json:"key,omitempty" description:"name of the environment variable; must be a C_IDENTIFIER; deprecated - use name instead"`
 	// Optional: defaults to "".
 	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
+	// Optional: specify a source for the value of the EnvVar.
+	Source *EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is not empty"`
+}
+
+// EnvVarSource represents a source for the value of an EnvVar.
+// Only one of its members may be specified.
+type EnvVarSource struct {
+	// Selects a field of the pod; only name and namespace are supported.
+	PodField *ObjectFieldSelector `json:"podField,omitempty" description:"selectes a field of the pod; only name and namespace are supported"`
+}
+
+// ObjectFieldSelector selects an APIVersioned field of an object.
+type ObjectFieldSelector struct {
+	// The API version the FieldPath is written in terms of.
+	APIVersion string `json:"apiVersion,omitempty" description="The API version that FieldPath is written in terms of"`
```

Comment:
> On Apr 21, 2015, at 6:28 PM, Tim Hockin <notifications@github.com> wrote:
> 
> In pkg/api/v1beta1/types.go:
> 
> > @@ -397,6 +397,23 @@ type EnvVar struct {
> >  	Key  string `json:"key,omitempty" description:"name of the environment variable; must be a C_IDENTIFIER; deprecated - use name instead"`
> >  	// Optional: defaults to "".
> >  	Value string `json:"value,omitempty" description:"value of the environment variable; defaults to empty string"`
> > +	// Optional: specify a source for the value of the EnvVar.
> > +	Source *EnvVarSource `json:"source,omitempty" description:"source for the environment variable's value; cannot be used if value is not empty"`
> > +}
> > +
> > +// EnvVarSource represents a source for the value of an EnvVar.
> > +// Only one of its members may be specified.
> > +type EnvVarSource struct {
> > +	// Selects a field of the pod; only name and namespace are supported.
> > +	PodField *ObjectFieldSelector `json:"podField,omitempty" description:"selectes a field of the pod; only name and namespace are supported"`
> > +}
> > +
> > +// ObjectFieldSelector selects an APIVersioned field of an object.
> > +type ObjectFieldSelector struct {
> > +	// The API version the FieldPath is written in terms of.
> > +	APIVersion string `json:"apiVersion,omitempty" description="The API version that FieldPath is written in terms of"`
> This is fine for now - it's not your fault/problem, but I want to be on record that asking the user to spec an API version inside an object that is already API-versioned is asinine. We should endeavor to do better than this in API v2.
> 
All we have to do is default it in the conversion.  We just don't have a cheap way of doing that.
> —
> Reply to this email directly or view it on GitHub.
>